docs: Simplify undestanding what's going on (#750)
antonbabenko/pre-commit-terraform: .github/CONTRIBUTING.md

### Finish with the documentation

1. Add the hook description to [Available Hooks](../README.md#available-hooks).
2. Create and populate a new hook section in [Hooks usage notes and examples](../README.md#hooks-usage-notes-and-examples).

> ### Work with Python hooks

Reviewer: Technically, `tox` wraps running tests with `pytest`, but it also is a command runner for other processes, like running `pre-commit. Stating that it's just Python would be misleading. Besides, running non-Python tests should probably be wrapped with either `pytest` or `tox` directly, making it the unified interface for testing it all the same way across local and CI envs.

Moreover, the word “hooks” is just as misleading because (1) it's a concept of the pre-commit framework while there are other things happening, like (2) running the tests and (3) linters.

Ideally, other processes would also migrate into tox subcommands.

That said, this could be temporarily titled as "Contributing to Python code" or something along the lines.

1. [Install `tox`](https://tox.wiki/en/stable/installation.html)
2. To run tests, run:

    ```bash
    tox -qq
    ```

>     If there are any issues, copy-paste and run the `python3 ...` command to visualize the pytest coverage report.

Reviewer 2: ```suggestion
    If there are any issues, copy-paste and run the failed `python3 ...` command to visualize the pytest coverage report.
```
Not sure whether "failed" is correct word in this context. Just want to give a reader a hint on what exactly needs to be copied and run. Please correct as applicable.

Author: It allays showed, no matter of execution result

Here is example:

```bash
➜ tox -qq                                         
==================================== test session starts ====================================
platform linux -- Python 3.10.12, pytest-8.3.4, pluggy-1.5.0
cachedir: .tox/py/.pytest_cache
rootdir: /home/vm/code/0-other/open-source/pre-commit-terraform
configfile: pytest.ini
testpaths: tests/pytest/
plugins: cov-6.0.0, xdist-3.6.1, mock-3.14.0
collected 11 items                                                                          

tests/pytest/_cli_test.py ....                                                        [ 36%]
tests/pytest/terraform_docs_replace_test.py .......                                   [100%]

---------- coverage: platform linux, python 3.10.12-final-0 ----------
Coverage HTML written to dir /home/vm/code/0-other/open-source/pre-commit-terraform/.tox/py/tmp/htmlcov/

Required test coverage of 100% reached. Total coverage: 100.00%

=================================== slowest 10 durations ====================================
0.01s setup    tests/pytest/_cli_test.py::test_known_interrupts[app-runtime-exc]

(9 durations < 0.005s hidden.  Use -vv to show these durations.)
==================================== 11 passed in 0.18s =====================================

To open the HTML coverage report, run

        python3 -Im webbrowser file:///home/vm/code/0-other/open-source/pre-commit-terraform/.tox/py/tmp/htmlcov/index.html

To serve the HTML coverage report with a local web server, use

        python3 -Im http.server --directory cov_html_report_dir 0

➜ 
```

Reviewer 2: Ah, I see. The maybe either not "failed" but "questionable" or not "`python3 ...`" but "``python3 -Im ...`"? 🤔

Reviewer 2: Or "copy-paste and run the `python3 ...` command from `tox -qq` output" 🤔

Reviewer: The coverage HTML report is always produced by default locally. Unless one overrides the pytest posargs when calling tox. Regardless of whether it's a failure or a success. It's just a convenient one-liner to open a coverage report view in one's browser as opposed to doing this manually.

3. Before committing any changes (if you do not have `pre-commit` installed locally), run:

    ```bash
    tox r -qq -e pre-commit
    ```

    Make sure that all checks pass.

> 4. (Optional): If you want to see more explicit MyPy checks, you can run:

Reviewer 2: ```suggestion
4. (Optional): If you want to see more details on MyPy checks, you can run:
```

> 4. (Optional): If you want to see more details on MyPy checks, you can run:

Reviewer: It doesn't give out more details. The coverage hints are always being printed out.
```suggestion
4. (Optional): If you want to limit the checks to MyPy only, you can run:
```

    ```bash
    tox r -qq -e pre-commit -- mypy --all-files
    ```

>     Then copy-paste and run the `python3 ...` commands to check the strictest MyPy coverage reports.

Reviewer: This is not an automated check, it's just a convenience command for opening an HTML file in a browser.
```suggestion
    Then copy-paste and run the `python3 ...` commands to inspect the strictest MyPy coverage reports visually.
```

> 5. (Optional): You can find all available environments by running:

Reviewer 2: ```suggestion
5. (Optional): You can find all available `tox` environments by running:
```
